agent/src/main/java/org/openjdk/jmc/agent/jfrnext/impl/JFRNextEventClassGenerator.java

```diff
@@ -231,6 +231,11 @@ private static void generateClassAnnotations(ClassWriter cw, JFRTransformDescrip
 		av.visit("value", td.getEventName());
 		av.visitEnd();
 
+		// Name
+		av = cw.visitAnnotation("Ljdk/jfr/Name;", true);
+		av.visit("value", TypeUtils.deriveIdentifierPart(td.getEventName()) + "Event");
```

Comment:
@thegreystone, reading your comment in #171, should we actually derive the name from the id then? I was wondering about that, too.